### PR TITLE
Feature

### PR DESCRIPTION
**Pull request by jmjoary**
- Add missing filters
- Add default property avatar for DirectusUser
- Add more flexibility on getDirectusUserList query result
- Add field property on updateDirectusUser to allow to retrieve the requested fields. * is by default.
- Directus Data can have id as an integer.
- Directus Data updateProperties must be set only if necessary. Otherwise, needSaving can false positive

### CODE REVIEW

**Review by maxbritto — COMMENTED** [review]
Is it me or do we have the exact same code twice at 2 different places ? 🤔

File: lib/src/directus_api_manager_base.dart [hunk 1]
```diff
@@ -190,14 +200,18 @@ class DirectusApiManager {
             roleUUID: roleUUID,
             otherProperties: otherProperties),
         parseResponse: (response) {
-          final DirectusItemCreationResult<Type> creationResult =
-              DirectusItemCreationResult(
-                  isSuccess:
-                      response.statusCode == 200 || response.statusCode == 204);
-          if (response.statusCode != 204) {
+          final DirectusItemCreationResult<Type> creationResult;
```
Comment (maxbritto) [inline lib/src/directus_api_manager_base.dart:203]: Duplicate code

File: lib/src/directus_api_manager_base.dart [hunk 2]
```diff
@@ -258,15 +272,18 @@ class DirectusApiManager {
             objectToCreate.endpointName, objectToCreate.toMap(),
             fields: fields),
         parseResponse: (response) {
-          final DirectusItemCreationResult<Type> creationResult =
-              DirectusItemCreationResult(
-                  isSuccess:
-                      response.statusCode == 200 || response.statusCode == 204);
-
+          final DirectusItemCreationResult<Type> creationResult;
```
Comment (maxbritto) [inline lib/src/directus_api_manager_base.dart:275]: Duplicate code

**Review by maxbritto — COMMENTED** [review]
Nice patch, I like the new filters added!

It would be nice to find a solution to avoid the code duplication for the result parsing (see both comments)

**Comment by jmjoary** [comment]
You are riche. I have notice it. You will get an update soon.

**Comment by maxbritto** [comment]
> You are riche
I'm not, but maybe I'm right instead 😁

**Comment by jmjoary** [comment]
I have removed the duplicate code

> > You are riche
> 
> I'm not, but maybe I'm right instead 😁

arf, auto correct issue on my phone.

**Review by maxbritto — APPROVED** [review]
Nice 👍

**Pull request merged**